packages/widget.vite-plugin/src/common/validateWidgetConfig.ts

```diff
@@ -75,5 +75,15 @@ function validateWidgetParameters(parameters: ParameterConfig): void {
         `Parameter id "${parameterId}" does not match allowed pattern (must be camelCase)`,
       );
     }
+    if (parameterConfig.type === "objectSet") {
+      if (
+        typeof parameterConfig.objectType.experimentalDoNotUseMetadata?.rid
```

Comment:
Just adding to the PR as well that at whatever point we announce this to users then we may be locking ourselves into experimentalDoNotUseMetadata to not break library versions that users depend on